backend/generate-db.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"fmt"
+	"gorm.io/driver/sqlite"
+	"gorm.io/gorm"
+	"os"
+)
+
+type Course struct {
+	ID   uint   `gorm:"primaryKey" json:"-"`
+	Code string `gorm:"unique,not null" json:"code"`
+	Name string `gorm:"not null" json:"name"`
+}
+
+type Tutor struct {
+	ID       uint   `gorm:"primaryKey" json:"-"`
+	Username string `gorm:"unique,not null" json:"username"`
+}
+
+type Tutoring struct {
+	TutorID  uint   `json:"-"`
+	Tutor    Tutor  `gorm:"foreignKey:TutorID" json:"tutor"`
+	CourseID uint   `json:"-"`
+	Course   Course `gorm:"foreignKey:CourseID" json:"course"`
+}
+
+func main() {
+	err := os.Remove("database.db") //remove old database, if needed
+	if err == nil {
+		fmt.Println("Deleting old database")
+	}
+	
+	fmt.Println("Creating new database")
+	db, _ := gorm.Open(sqlite.Open("database.db"), &gorm.Config{}) //create new db
+	_ = db.AutoMigrate(&Course{}, &Tutor{}, &Tutoring{}) //set up schema
+
+	fmt.Println("Populating database")
+	
+	//add tuples for courses
```

Comment:
Slight nitpick, these aren't tuples since they're unnamed, it's a struct literal. I also don't this comment really adds anything, it should be clear from the code.

Reply:
Removed it. Will push changes.